std/uni.d

```diff
@@ -2318,7 +2318,7 @@ public:
      * $(LI $(B %x) formats the intervals as a [low..high$(RPAREN) range of lowercase hex characters)
      * $(LI $(B %X) formats the intervals as a [low..high$(RPAREN) range of uppercase hex characters)
      */
-    void toString(scope void delegate(const(char)[]) sink,
+    void toString(Writer)(scope Writer sink,
```

Comment:
I did it to match what is done in `std.format`, which uses `Writer` with no constraints. As an improvement I think there should be an `isWriter!Writer` template defined somewhere, because this will come up again and again, and then all the sinkers switch to it, but considered it outside the scope of this PR.